clc/modules/block-storage/src/main/java/com/eucalyptus/blockstorage/S3SnapshotTransfer.java

```diff
@@ -272,13 +272,14 @@ public String prepareForUpload() throws SnapshotTransferException {
       Path zipFilePath = Files.createTempFile(Paths.get("/var/tmp"), keyName + '-', '-' + String.valueOf(partNumber));
       part = SnapshotPart.createPart(snapUploadInfo, zipFilePath.toString(), partNumber, readOffset);
 
-      InputStream inputStream = storageResource.getInputStream();
       ByteArrayOutputStream baos = new ByteArrayOutputStream();
       GZIPOutputStream gzipStream = new GZIPOutputStream(baos);
       FileOutputStream outputStream = new FileOutputStream(zipFilePath.toString());
+      InputStream inputStream = null;
 
       try {
         LOG.debug("Reading snapshot " + snapshotId + " and compressing it to disk in chunks of size " + partSize + " bytes or greater");
+        inputStream = storageResource.getInputStream( );
```

Comment:
This is the snapshot transfer resource handling fix, ensures stream cleaned up if open